common/djangoapps/student/views.py

```diff
@@ -1644,6 +1646,11 @@ def create_account_with_params(request, params):
     if should_link_with_social_auth or (third_party_auth.is_enabled() and pipeline.running(request)):
         params["password"] = pipeline.make_random_password()
 
+    if active_provider_requests_data_sharing(request):
+        extra_fields['data_sharing_consent'] = 'optional'
+    if active_provider_enforces_data_sharing(request, EnterpriseCustomer.AT_LOGIN):
+        extra_fields['data_sharing_consent'] = 'required'
```

Comment:
No, unfortunately not.

Basically, the standard registration flow takes advantage of two levels of form validation. First, there's the basic UI-level validation that happens in `user_api/views.py`; it constructs a form descriptor that defines what's required in a given form.

If the data entered passes that form, then the data (and only the data; none of the form description) is passed into `student/views.py`, which does some manipulation and then uses `student.forms.py:AccountCreationForm` to do a full validation pass.

The trouble comes in because `student/views.py` is accessible independent of `user_api/views.py`; it can be used to create a user without any prior UI-level validation.

Basically, we need validation in both places. This particular place could certainly use some cleanup though.